### DISCUSSION TITLE
Personal Access Token deleting itself

### DISCUSSION THREAD

**Discussion by battlesquid** [community/community#24692] [Apps, API and Webhooks]
<p>I set up a <a href="https://battlesquid.github.io" rel="nofollow noopener">website </a>using GH pages to showcase some of my repos. I had tested previously on <a href="https://repl.it" rel="nofollow noopener">repl.it </a>with no issues, and I was ready to migrate to Github. However, I found that my personal access token had deleted itself when my page wanted to grab the repo info (similar to <a href="https://stackoverflow.com/questions/53579650/github-api-personal-access-token-removes-itself" rel="nofollow noopener">this</a> problem). Here’s the code I use to fetch the repo data.</p>
<pre><code>const query = {
  query: "query {user(login:\"Battlesquid\") {bio,repositories(isFork:false, first:100) {nodes {repositoryTopics(first:5) {nodes {topic {name}}}name,description, primaryLanguage {name,color}, languages(first:100) {nodes {name}}}}}}"
};

const res = await fetch("https://api.github.com/graphql", {
    method: "post",
    headers: {
      "Content-Type": "application/json",
      "Authorization": "Bearer &lt;TOKEN&gt;",
      "User-Agent": "Battlesquid"
    },
    body: JSON.stringify(query)
  })
</code></pre>
<p>How can I fix this issue? Thanks in advance!</p>

**Comment by yjose** [comment, marked as answer]
<p>Probably you commit the token to your GitHub repo. if its the case Github will remove the token immediately. so you need to set your token as a variable environment in your system.</p>

**Comment by rhysd** [comment]
<p>Personal access token which has not been used for one year is automatically deleted for security reason.</p>
<p>See: <a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" rel="noopener nofollow ugc">https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token</a></p>
<blockquote>
<p>As a security precaution, GitHub automatically removes personal access tokens that haven’t been used in a year.</p>
</blockquote>
